### ISSUE TITLE
Drag and drop does not work in reaper Linux.

### ISSUE THREAD

**Issue by TheGuyWhoo** [surge-synthesizer/surge#6700]
Reaper v6.64, linux mint 20 64 bit. surge xt 1.1.2. Present in both vst3 and lv2.

Trying to drag and drop a file into wavetable sample does not work. Nothing happens, surge doesn't seem to see the sample at all as the sample is actually loaded into the project when you drag the sample on to surge.

**Comment by baconpaul** [comment]
I don't know if i have ever tested drag and drop in reaper linux! I just tried it on macOS and it worked so I changed the title of the issue.

**Comment by TheGuyWhoo** [comment]
I will quickly download ardour and see if the issue persists there.

**Comment by baconpaul** [comment]
the surge standalone could also be a useful test
i know there's something about xembed and drag and drop and so on that honestly i've never figured out
there is a menu item to load a wavetable directly in the wavetable menu i'm pretty sure if dnd doesn't work

**Comment by TheGuyWhoo** [comment]
Drag and drop works in the standalone.

**Comment by baconpaul** [comment]
right so this sounds like a reaper embedding problem then. hmm. lemme think about how to report this to them.

you dont' run reaper dev right? So you can't try the clap.

**Comment by TheGuyWhoo** [comment]
No I'm running a regular build. Is reaper dev available to the public? I don't see it on the download page.

**Comment by baconpaul** [comment]
oh i thought it was. but maybe it isn't! @mkruselj will know.

**Comment by mkruselj** [comment]
Reaper AFAIK doesn't do drag&drop of media from the timeline into the plugins. Not a Surge issue.

**Comment by TheGuyWhoo** [comment]
I'm actually not sure why you're assuming that I'm trying to do it from the timeline, I'm actually trying to do it from the media explorer. Drag and drop of any kind is not working.

**Comment by mkruselj** [comment]
Works here on Windows, so this would be a Reaper in Linux issue.

**Comment by TheGuyWhoo** [comment]
Hmm I don't seem to recall trying it with other plugins yet... Do you happen to know of other linux vsts that support drag and drop so i

**Comment by TheGuyWhoo** [comment]
Sorry accidentally hit close

**Comment by mkruselj** [comment]
I mean it can be closed since it's not a Surge issue, since drag&drop from Reaper's Media Explorer works here on Win (I'd bet it also works fine on Mac too).

Seems some other plugins have a similar issue too: https://www.kvraudio.com/forum/viewtopic.php?t=570716

**Comment by TheGuyWhoo** [comment]
I remember now I have tried making a sampler with cabbage and drag and drop didn't work. It likely is an issue with reaper.

**Comment by baconpaul** [comment]
yeah especially since it works in the surge standalone

I think you probably need to report this on the reaper forums. If you do feel free to tag that over here and I can chime in with the code path we use to support it (we really are just a JUCE drop target)

drag and drop (along with other standardized gestures) on linux is not as easy as on mac or windows, because of the multiple ways to set up window managers etc.... and I know Reaper uses a pretty low level windowing API. I wonder if they are just not supporting the right embed message in their plugin window.

Reaper has a way to run outside of their plugin window right? Like a "not wrapped" mode or something? I wonder if in that mode surge would work.

**Comment by TheGuyWhoo** [comment]
https://forum.cockos.com/showthread.php?p=2613293#post2613293

**Comment by TheGuyWhoo** [comment]
Oh and I think you're referring to the feature that let you run the plugin as a separate process? I just tried that and the behavior does not change, reaper still eats the drop.

**Comment by TheGuyWhoo** [comment]
Cool if I close this? It's definitely a reaper issue.

**Comment by Andreya-Autumn** [comment]
Done. ;)

**Comment by TheGuyWhoo** [comment]
It is a juce problem

https://forum.juce.com/t/drag-and-drop-not-working-under-vst-in-linux/50410

**Comment by haenkel** [comment]
> It is a juce problem

yes ...but not in this case.
This was for the bespoke issue you've opened
https://github.com/BespokeSynth/BespokeSynth/issues/970
(bespoke is a juce host, reaper is not, dnd works in ardour etc.)

**Comment by TheGuyWhoo** [comment]
Would it not also apply to juce plugins?

**Comment by baconpaul** [comment]
If you read that thread it looks more like a “daw owns the parent window and x doesn’t let you consistently inject xdnd into the hierarchy in every daw” - I’m not sure what a solution would be for either a juce or a non juce plugin that considered right?

**Comment by TheGuyWhoo** [comment]
Welp I just tried in ardour and drag and drop does not work for me. Honestly if I may suggest an alternative, in reaper copying the full filepath is trivial. The ability to load a sample directly from a filepath would be a big improvement for me.

**Comment by haenkel** [comment]
click on wt name and from up popping menu choose "Load Wavetable From File..." 
opens a file browser that lets you do that

**Comment by TheGuyWhoo** [comment]
Yes that works everywhere, that is how I've been doing it. Massive pain in the ass though, frankly.

**Comment by TheGuyWhoo** [comment]
Oh wait I see what you mean. Didn't know you could do that there, thank you!
